src/store/modules/order/actions.ts

```diff
@@ -219,7 +219,122 @@ const actions: ActionTree<OrderState, RootState> = {
     }
 
     return payload;
+  },
+
+  updateCurrent ({ commit }, payload) {
+    commit(types.ORDER_CURRENT_UPDATED, { order: payload.order })
+  },
+
+  async fetchStuckOrders({commit}){
+    let resp;
+    const payload = {
+      "json": {
+        "params": {
+          "rows": "20",
+          "sort": "orderDate asc"
+        },
+        "query":"(*:*)",
+        "filter": ["docType:ORDER","orderTypeId: SALES_ORDER","facilityId: _NA_","orderStatusId: ORDER_APPROVED" ]
+      }
+    }
+    try {
+      resp = await OrderService.findOrderDetails(payload);
+      if (resp.status === 200 && !hasError(resp)) {
+        const productIds = resp.data.response.docs.map((order: any) => {
+          return order.productId;
+        })
+        const orderIds = resp.data.response.docs.map((order: any) => {
+          return order.orderId;
+        })
+        const viewSize = productIds.length;
+        const viewIndex = 0;
+        const payload = {
+          viewSize,
+          viewIndex,
+          productIds
+        }
+        await this.dispatch('stock/addProducts', payload);
+        await this.dispatch("product/fetchProducts", payload);
```

Comment:
We should fetch this information and add it to item object to be use directly instead of using getters